File: code/bit_manipulation/src/count_set_bits/count_set_bits.cpp
#include <iostream>
using namespace std;
// Part of Cosmos by OpenGenus Foundation
int count(int n)
{
    int c = 0;
    while (n)
    {
        c++;
        n &= (n - 1);
    }
    return c;
}

int main()
{

    int n;
    cin >> n;

    #ifdef BUILTIN
    cout << __builtin_popcount(n); // use builtin popcount
    #else
    cout << count(n);  // manual
    #endif

    return 0;
}

File: code/bit_manipulation/src/count_set_bits/count_set_bits.js
/*

// bit manipulation | count set bits | count set bits | JAVASCRIPT
// Part of Cosmos by OpenGenus Foundation

*/


// Part of Cosmos by OpenGenus Foundation

export default function countSetBits(n) {
  n = +n;

  if (!(n >= 0 || n < 0)) {
    return 0;
  } else if (n < 0) {
    throw new Error("negtive numbers are not supported");
  }

  let cnt = 0;
  while (n) {
    cnt++;
    n &= n - 1;
  }

  return cnt;
}
